sintr_analysis_server_example/worker_isolate.dart

```diff
@@ -37,61 +32,36 @@ Future<String> _protectedHandle(String msg) async {
     var inputData = JSON.decode(msg);
     String bucketName = inputData[0];
     String objectPath = inputData[1];
-    var logItems = [];
+    var results = [];
     var errItems = [];
     int failureCount = 0;
     int lines = 0;
 
-    LogItemProcessor proc = new LogItemProcessor(completionExtraction);
+    // Initialize query specific objects
+    var mapper = new VersionMapper();
 
     Stream dataStream = await getDataFromCloud(bucketName, objectPath);
-    await for (String s in dataStream) {
-      lines++;
-      try {
-        proc.addRawLine(s);
-
-        String nextMessage;
-        while (proc.hasMoreMessages) {
-          nextMessage = proc.readNextMessage();
-          if (nextMessage != null) logItems.add(nextMessage);
-        }
-      } catch (e, st) {
-        log.info("Message proc erred. $e \n $st \n");
-        var rawErringLine = s.toString();
-        if (rawErringLine.length > 300) rawErringLine = '${rawErringLine.substring(0, 300)} ...';
-        errItems
-            .add({"lastBlock": false, "rawDataLine": rawErringLine, "exception": "$e", "stackTrace": "$st"});
-        failureCount++;
-      }
-    }
-
-    if (client != null) client.close();
-
-    try {
-      proc.close();
 
-      String nextMessage;
-      while (proc.hasMoreMessages) {
-        nextMessage = proc.readNextMessage();
-        if (nextMessage != null) logItems.add(nextMessage);
-      }
-    } catch (e, st) {
-      log.info("Message proc erred. $e \n $st \n");
-      errItems
-          .add({"lastBlock": true, "exception": "$e", "stackTrace": "$st"});
+    // Extraction
+    await mapper.init({});
+    await for (String logEntry in dataStream
+        .transform(UTF8.decoder)
+        .transform(new LineSplitter())
+        .transform(new LogItemTransformer())
+        .handleError((e, s) {
       failureCount++;
+      errItems.add("Error reading line\n${trim300(e.toString())}\n$s");
+    })) {
+      lines++;
+      var result = mapper.map(logEntry);
```

Comment:
Good point, it's inside a larger try catch block already from ln 31, but it could probably do with more localised error handling as well.